src/dbally/data_models/llm_options.py

```diff
@@ -1,6 +1,8 @@
 from dataclasses import asdict, dataclass
 from typing import List, Optional, Union
 
+from openai import NOT_GIVEN, NotGiven
```

Comment:
By doing this we force the user to download openai module even though they want to use different LLM provider. 

Maybe we should use define these types by ourselves? https://github.com/openai/openai-python/blob/main/src/openai/_types.py#L107

Reply:
We can't define a custom sentinel here because it can't be parsed by the OpenAI client, LLMOptions are tied to a specific LLM client, and each integration will require us to write separate options. This is related to the fact that each LLM client has a slightly different API and requires different parameters to be set, so we need this separation anyway.

Reply:
I literally tried this earlier and received an error from the openai module.

Reply:
A nasty solution would be to convert these inside OpenAIClient.call

Reply:
I would leave it in the `dbally.data_models.llm_options` module and only rename the file and create a base class for next integrations. What do you think?

cc @mhordynski @ludwiktrammer